### PR TITLE
Added Order and his dao and service

### PR DESCRIPTION
**Pull request by vicire**

### CODE REVIEW

File: src/main/java/cinema/dao/OrderDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Order> getOrderHistory(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<Order> orderHistory = session.createQuery("from Order o "
```
Comment (Max-Pochepets) [inline src/main/java/cinema/dao/OrderDaoImpl.java]:
```suggestion
            Query<Order> orderHistory = session.createQuery("from Order "
```
redundant alias

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/cinema/dao/OrderDaoImpl.java (hunk 1 above)
Comment (vicire) [reply src/main/java/cinema/dao/OrderDaoImpl.java]: At this time it`s not redundant :)

File: src/main/java/cinema/Main.java [hunk 2]
```diff
@@ -22,6 +23,10 @@ public static void main(String[] args) {
         movie.setTitle("Fast and Furious");
         MovieService movieService = (MovieService) injector.getInstance(MovieService.class);
         movieService.add(movie);
+        Movie movie1 = new Movie();
+        movie1.setTitle("Lord of the rings");
+        movie1.setDescription("hobbits and elfs");
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/Main.java:28]: Orks and Humans: Are we joke for you?

File: src/main/java/cinema/dao/OrderDaoImpl.java (hunk 1 above)
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/dao/OrderDaoImpl.java]: https://mateacademy.slack.com/archives/G01EHB8CWAE/p1612455330236200

File: src/main/java/cinema/model/Order.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToMany
+    private List<Ticket> tickets;
+    @Column(name = "order_date")
+    private LocalDateTime orderDate;
+    @OneToOne
+    private User user;
```
Comment (dmytro-hryhoruk) [inline src/main/java/cinema/model/Order.java]: user may have more than one Order

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

File: src/main/java/cinema/Main.java (hunk 2 above)
Comment (vicire) [reply src/main/java/cinema/Main.java:28]: Sauron lost. They are not honoured to be mentioned :)

File: src/main/java/cinema/dao/OrderDaoImpl.java (hunk 1 above)
Comment (vicire) [reply src/main/java/cinema/dao/OrderDaoImpl.java]: Fixed

File: src/main/java/cinema/model/Order.java (hunk 3 above)
Comment (vicire) [reply src/main/java/cinema/model/Order.java]: Fixed. But I have a question. We formed an order from the shopping cart. After that, we add order to the database and after we cleared our shopping cart. And if we choose another tickets, we will form another order. So, by one session we will form one order for one user. So why `ManyToOne` assignment prevailed over `OneToOne` in this case?

Comment (dmytro-hryhoruk) [reply src/main/java/cinema/model/Order.java]: It's doesn't relate to whether we are doing this during one session or not(if you are talking about hibernate session). This annotation  discribe the relations between entities. One user can have multiple orders;

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**